$NetBSD: patch-configure.ac,v 1.2 2013/10/03 23:16:17 makoto Exp $
The function to check giflib, EGifPutExtentionLast is no longer
available for gif 5.0.4, use ..Trailer instead

--- configure.ac.orig	2013-10-04 06:46:47.000000000 +0900
+++ configure.ac	2013-10-04 07:31:45.000000000 +0900
@@ -3097,6 +3097,7 @@
 ### Use -lgif or -lungif if available, unless `--with-gif=no'.
 ### mingw32 doesn't use -lgif/-lungif, since it loads the library dynamically.
 HAVE_GIF=no
+HAVE_GIF4=no
 LIBGIF=
 if test "${opsys}" = "mingw32"; then
   if test "${with_gif}" != "no"; then
@@ -3110,9 +3111,13 @@
   AC_CHECK_HEADER(gif_lib.h,
 # EGifPutExtensionLast only exists from version libungif-4.1.0b1.
 # Earlier versions can crash Emacs.
-    [AC_CHECK_LIB(gif, EGifPutExtensionLast, HAVE_GIF=yes, HAVE_GIF=maybe)])
+    [AC_CHECK_LIB(gif, EGifPutExtensionTrailer, HAVE_GIF=yes, HAVE_GIF=maybe)])
+  AC_CHECK_HEADER(gif_lib.h,
+    [AC_CHECK_LIB(gif, EGifPutExtensionLast,    HAVE_GIF4=yes, HAVE_GIF4=maybe)])
 
-  if test "$HAVE_GIF" = yes; then
+  if test "$HAVE_GIF4" = yes; then
+    LIBGIF=-lgif
+  elif test "$HAVE_GIF" = yes; then
     LIBGIF=-lgif
   elif test "$HAVE_GIF" = maybe; then
 # If gif_lib.h but no libgif, try libungif.
@@ -3123,6 +3128,9 @@
   if test "${HAVE_GIF}" = "yes"; then
     AC_DEFINE(HAVE_GIF, 1, [Define to 1 if you have a gif (or ungif) library.])
   fi
+  if test "${HAVE_GIF4}" = "yes"; then
+    AC_DEFINE(HAVE_GIF4, 1, [Define to 1 if you have a gif (or ungif) library.])
+  fi
 fi
 AC_SUBST(LIBGIF)
 
@@ -3137,6 +3145,7 @@
   test "${with_png}" != "no" && test "${HAVE_PNG}" != "yes" &&
     MISSING="$MISSING libpng" && WITH_NO="$WITH_NO --with-png=no"
   test "${with_gif}" != "no" && test "${HAVE_GIF}" != "yes" &&
+                                test "${HAVE_GIF4}" != "yes" &&
     MISSING="$MISSING libgif/libungif" && WITH_NO="$WITH_NO --with-gif=no"
   test "${with_tiff}" != "no" && test "${HAVE_TIFF}" != "yes" &&
     MISSING="$MISSING libtiff" && WITH_NO="$WITH_NO --with-tiff=no"
